File: ride-service/src/main/java/com/example/rideservice/service/PriceService.java
package com.example.rideservice.service;

import com.example.rideservice.dto.request.PriceCalculateRequest;
import com.example.rideservice.dto.response.PriceCalculateResponse;

public interface PriceService {

    PriceCalculateResponse getCalculatePrice(PriceCalculateRequest request);
}

File: price-calculator-service/src/main/kotlin/org/example/pricecalculatorservice/service/PriceService.kt
package org.example.pricecalculatorservice.service

import org.example.pricecalculatorservice.dto.PriceCalculateRequest
import org.example.pricecalculatorservice.dto.PriceCalculateResponse
import org.springframework.security.oauth2.jwt.Jwt


interface PriceService {
    fun calculatePrice(priceRequest: PriceCalculateRequest): PriceCalculateResponse?
    fun extractUserInfo(jwt: Jwt): Any?
}
